packages/wrangler/src/api/pages/publish.tsx

```diff
@@ -258,17 +305,19 @@ export async function publish({
 				nodejsCompat,
 			});
 		} else {
-			await checkRawWorker(workerScriptPath, () => {});
-			// TODO: Replace this with the cool new no-bundle stuff when that lands: https://github.com/cloudflare/workers-sdk/pull/2769
+			await checkRawWorker(_workerPath, () => {});
+			// TODO: Let users configure this in the future.
 			workerBundle = {
 				modules: [],
 				dependencies: {},
 				stop: undefined,
-				resolvedEntryPointPath: workerScriptPath,
+				resolvedEntryPointPath: _workerPath,
 				bundleType: "esm",
 			};
 		}
+	}
 
+	if (_workerJS || _workerJSDirectory) {
 		const workerBundleContents = await createUploadWorkerBundleContents(
 			workerBundle as BundleResult
```

Comment:
Is this cast needed?

Reply:
Yes. Our logic here of `_workerJS`, `_workerJSDirectory` and `builtFunctions`, and how they can each be producing a `workerBundle` is a bit too complicated for TS to get it automatically. We've already got it on the other side too: https://github.com/cloudflare/workers-sdk/blob/6f5259f73e7a93258524786f817ca1186ae06bd4/packages/wrangler/src/api/pages/publish.tsx#L307